coding/react.md

```diff
@@ -6,13 +6,23 @@ layout: base
 
 This document outlines our standards for writing React applications.
 
-## Starter Project
+## Starting a new Project
+We recommend [create-react-app](https://github.com/facebook/create-react-app) for bootstrapping a new project, and avoid ejecting as long as feasible.
 
-[canonical-webteam/react-starter](https://github.com/canonical-webteam/react-starter) provides a codified
-example of our standards for react projects (Prettier, AirBnB style).
+### Preferred libraries
+* routing - [react-router](https://github.com/ReactTraining/react-router)
+* app state management - [redux](https://redux.js.org)
```

Comment:
I don't think we should mandate always using redux? Maybe state that if your application will contain complex state then redux is recommended.

Reply:
I don't read this section as 'mandating'. I guess the intension is just to give us suggestions in terms of: "does you app need state management - consider redux".

Same way not every react app will need routing, not everyone will need redux.

Reply:
It could be read either way, but I do think there's a subtle thread here that we want to consider.

I noticed that @squidsoup's previous PR on this document changed the wording from "react components" to "react applications". I was gonna comment on it but didn't get around to it before it was merged. The direction we seem to be moving here is towards suggesting that, for projects that use React, React is somehow the *essence* of the application rather than a *tool* used by the application.

For some applications, of course React is very central. But I agree with @anthonydillon that philosophically we should consider React, Redux, the React Router and any other pieces to be tools at our disposal rather than central parts of the framework of our applications. Each should be chosen and used based on its merits.

snapcraft.io is a good example, where we're pulling in pieces of React for certain pages. It definitely shouldn't be considered "a React application".

Reply:
> The direction we seem to be moving here is towards suggesting that, for projects that use React, React is somehow the essence of the application rather than a tool used by the application.

That sounds perfectly accurate to me. I can't speak for BSI, as I'm not familiar with the codebase, but RBAC is very much a _react_ application and is built using react design patterns (flux). React likes to tout itself as a view library, but it is very much a framework, albeit not a very opinionated one.

Reply:
@nottrobin I think reading through this diff again, the advice is equally applicable to both react applications (applications like BSI and RBAC that are entirely built in react), and applications that use partially react. The advice around routing and state management naturally just don't apply to the later.